src/WebJobs.Extensions.DurableTask.Scale/AzureStorage/AzureStorageScalabilityProviderFactory.cs

```diff
@@ -0,0 +1,223 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask.Scale.AzureStorage
+{
+    /// <summary>
+    /// Factory class responsible for creating <see cref="AzureStorageScalabilityProvider"/> instances.
+    /// </summary>
+    public class AzureStorageScalabilityProviderFactory : IScalabilityProviderFactory
+    {
+        private const string LoggerName = "Host.Triggers.DurableTask.AzureStorage";
+        internal const string ProviderName = "AzureStorage";
+
+        private readonly DurableTaskScaleOptions options;
+        private readonly IStorageServiceClientProviderFactory clientProviderFactory;
+        private readonly INameResolver nameResolver;
+        private readonly ILoggerFactory loggerFactory;
+        private AzureStorageScalabilityProvider defaultStorageProvider;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AzureStorageScalabilityProviderFactory"/> class.
+        /// </summary>
+        /// <param name="options">The durable task scale options.</param>
+        /// <param name="clientProviderFactory">The storage client provider factory.</param>
+        /// <param name="nameResolver">The name resolver for connection strings.</param>
+        /// <param name="loggerFactory">The logger factory.</param>
+        /// <exception cref="ArgumentNullException">Thrown when required parameters are null.</exception>
+        public AzureStorageScalabilityProviderFactory(
+            IOptions<DurableTaskScaleOptions> options,
+            IStorageServiceClientProviderFactory clientProviderFactory,
+            INameResolver nameResolver,
+            ILoggerFactory loggerFactory)
+        {
+            this.clientProviderFactory = clientProviderFactory ?? throw new ArgumentNullException(nameof(clientProviderFactory));
+            this.nameResolver = nameResolver ?? throw new ArgumentNullException(nameof(nameResolver));
+            this.loggerFactory = loggerFactory ?? throw new ArgumentNullException(nameof(loggerFactory));
+
+            var optionsValue = options?.Value ?? throw new ArgumentNullException(nameof(options));
+
+            // In Scale Controller context, optionsValue will be a default/empty object (can't read host.json)
+            // The real configuration comes from triggerMetadata in GetScalabilityProvider()
+            this.options = optionsValue;
```

Comment:
If this is default/empty as the comment says, why do we accept it as a parameter and save it?